staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -208,6 +212,23 @@ type ValidationRule struct {
 	// "x must be less than max ("+string(self.max)+")"
 	// +optional
 	MessageExpression string
+	// reason provides a machine-readable validation failure reason that is returned to the caller when a request fails this validation rule.
+	// The HTTP status code returned to the caller will match the reason of the reason of the first failed validation rule.
+	// The currently supported reasons are: "FieldValueInvalid", "FieldValueForbidden", "FieldValueNotFound", "FieldValueTooLong", "FieldValueRequired", "FieldValueDuplicate", "FieldValueNotSupported", "FieldValueTooMany".
+	// If not set, default to use "FieldValueInvalid".
+	// All future added reasons must be accepted by clients when reading this value.
+	// +optional
+	Reason *field.ErrorType
```

Comment:
This surprises that we use some util package types in APIs. Changes won't require api review.

Reply:
We have the allow enum list in API changes so any change will trigger the API review :)